File: Data-Structures-and-Algorithms/0237-DeleteNodeinaLinkedList/deleteNode.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution(object):
    def deleteNode(self, node):
        curr = node
        curr.val = curr.next.val
        curr.next = curr.next.next

LL = ListNode(0)
deletenode = LL.next = ListNode(1)
LL.next.next = ListNode(2)
LL.next.next.next = ListNode(3)
LL.next.next.next.next = ListNode(4)
LL.next.next.next.next.next = ListNode(5)
print(Solution().deleteNode(deletenode))
node = LL
while node:
    print(node.val)
    node = node.next

# O(1)
# O(1)
# Given a node, delete it from the linked list guranteed that it is not the last node

File: Data-Structures-and-Algorithms/0237-DeleteNodeinaLinkedList/deleteNode.js
function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

const deleteNode = function(node) {
    node.val = node.next.val;
    node.next = node.next.next;
};

let LL = new ListNode(0);
LL.next = new ListNode(1);
LL.next.next = new ListNode(2);
let deletenode = LL.next.next.next = new ListNode(3);
LL.next.next.next.next = new ListNode(4);
LL.next.next.next.next.next = new ListNode(5);
deleteNode(deletenode);
node = LL;
while (node) {
    console.log(node.val);
    node = node.next;
};
    
// O(1)
// O(1)
// Given a node, delete it from the linked list guranteed that it is not the last node